### ISSUE TITLE
implicit use of %cl by SHL32CL not propagated to *.uni

### ISSUE THREAD

**Issue by matsc-at-sics-se** [matsc-at-sics-se/unison#2]
shell:
```
uni import infback.mir -o infback.uni --target=X86 --maxblocksize=50 --goal=speed --copyremat --lint
allTemporariesUsed: t34 is not used
```
infback.mir, see %26-%28:
```
  bb.8 (%ir-block.30, freq 13):
    successors: %bb.9(100)
  
    %2 = COPY %24
    MOV64mr %4, 1, _, 56, _, %2, <0x524a090> = !{!"unison-memory-partition", i32 0} :: (store 8 into %ir.32)
    MOV32mi %2, 1, _, 20, _, 32768, <0x524a090> = !{!"unison-memory-partition", i32 0} :: (store 4 into %ir.34)
    MOV32mr %2, 1, _, 48, _, %5, <0x524a090> = !{!"unison-memory-partition", i32 0} :: (store 4 into %ir.36, align 8)
    %26 = COPY %5:sub_8bit
    %cl = COPY %26
    %28 = SHL32rCL %22, implicit-def dead %eflags, implicit %cl
    MOV32mr %2, 1, _, 52, _, killed %28, <0x524a090> = !{!"unison-memory-partition", i32 0} :: (store 4 into %ir.39)
    MOV64mr %2, 1, _, 64, _, %6, <0x524a090> = !{!"unison-memory-partition", i32 0} :: (store 8 into %ir.41)
    MOV64mi32 %2, 1, _, 56, _, 0, <0x524a090> = !{!"unison-memory-partition", i32 0} :: (store 8 into %ir.45)
    %25 = MOV32r0 implicit-def dead %eflags
```
infback.uni, see o66-o67:
```
b11 (freq: 13):
    o62: [] <- (in) []
    o63: [] <- MOV64mr [t0,1,_,56,_,t24] (mem: 0)
    o64: [] <- MOV32mi [t24,1,_,20,_,32768] (mem: 0)
    o65: [] <- MOV32mr [t24,1,_,48,_,t1] (mem: 0)
    o66: [t34] <- (low) [t1]
    o67: [t35] <- SHL32rCL [t22]
    o68: [] <- MOV32mr [t24,1,_,52,_,t35] (mem: 0)
    o69: [] <- MOV64mr [t24,1,_,64,_,t2] (mem: 0)
    o70: [] <- MOV64mi32 [t24,1,_,56,_,0] (mem: 0)
    o71: [t36] <- MOV32r0 []
    o72: [] <- (out) []
[infback.mir.txt](https://github.com/matsc-at-sics-se/unison/files/1763758/infback.mir.txt)

```

**Comment by robcasloz** [comment]
Hi, I cannot reproduce this one, perhaps you have local changes to be pushed?

**Comment by matsc-at-sics-se** [comment]
Maybe, but they should not affect this. Can you send me your infback.uni?

**Comment by robcasloz** [comment]
My infback.uni is attached ([infback.uni.txt](https://github.com/matsc-at-sics-se/unison/files/1766109/infback.uni.txt)), looks like the temporaries are offset by one:

uni import infback.mir -o infback.uni --target=X86 --maxblocksize=50 --goal=speed --copyremat --lint
allTemporariesUsed: t33 is not used
(...)

**Comment by matsc-at-sics-se** [comment]
Right, t33 is defined but not used in your version (t34 in my version).
I have pushed my changes.

**Comment by matsc-at-sics-se** [comment]
Fixed by commit 3b46831b26b8f375f15deed07a3b05a616e0b9e1.
